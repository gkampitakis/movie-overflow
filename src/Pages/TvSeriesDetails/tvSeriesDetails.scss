.tv_series_details {
  display: flex;
  width: 70%;
  margin: 100px auto;
  padding-bottom: 40px;

  .details {
    height: 100%;
    flex: 3;
    box-shadow: 0 2px 8px rgba(0, 0, 0, 0.1);
    border-bottom-right-radius: 8px;
    border-bottom-left-radius: 8px;

    .tv_header {
      background: #333;
      color: white;
      height: 100px;
      justify-content: space-between;
      display: flex;
      flex-direction: column;
      border-top-left-radius: 8px;
      border-top-right-radius: 8px;

      h1 {
        font-size: 1.5rem;
        letter-spacing: 1px;
        margin: 5px;
      }

      p {
        margin: 5px;
        font-size: 1rem;
      }
    }

    .overview {
      display: flex;
      flex-direction: row;

      img {
        max-width: 200px;
      }

      div {
        .tagline {
          font-style: italic;
          text-align: center;
          font-weight: 600;
        }

        p {
          font-size: 0.9rem;
          text-align: left;
          margin: 10px;
        }

        display: flex;
        vertical-align: middle;
        flex-direction: column;
        justify-content: space-around;
      }
    }

    hr {
      margin: 10px 0;
      border: 0.1px solid #e6e6e6;
    }

    .general {
      margin: 0.35em 0 0.25em;
      color: #666666;
      font-size: 13px;
      margin: 10px;
      justify-content: space-around;
      display: flex;

      h3 {
        padding: 0;
        font-weight: bold;
      }

      .released {
        justify-content: space-evenly;
        text-align: center;
        display: flex;
        width: 50%;
      }

      .statistics {
        justify-content: space-around;
        display: flex;
        flex-direction: row;
      }
    }

    .seasons_title {
      font-size: 1.2rem;
      font-weight: 600;
      margin: 10px;
    }

    .seasons {
      display: flex;
      overflow: auto;
      width: 50vw;
      margin: auto 20px;

      .season {
        cursor: pointer;
        transition: all linear 0.2s;
        margin: auto 20px;

        img {
          max-width: 100px;
        }

        h3 {
          font-weight: bold;
          color: #666666;
          text-align: center;
        }

        &.selected {
          background: red;
        }

        &:first-child {
          margin-left: unset;
        }
      }
    }

    .season_details {
      margin: 10px;
      h2 {
        font-size: 1.1rem;
        font-weight: bold;
        margin-bottom: 10px;
      }
    }
  }

  .credits {
    text-align: center;
    flex: 1;
    height: 69vh;
    overflow: auto;
    min-width: 350px;

    h2 {
      font-size: 1.5rem;
      letter-spacing: 1px;
    }

    div {
      box-shadow: 0 2px 8px rgba(0, 0, 0, 0.1);
      width: 50%;
      margin: 10px auto;
      border-radius: 8px;
      cursor: pointer;

      h3 {
        font-size: 1rem;
        font-weight: 500;
      }
      p {
        color: #666666;
      }
    }

    .avatar {
      width: 130px;
      height: 130px;
      margin: 25px auto;
      overflow: hidden;
      border-radius: 50%;
      position: relative;

      img {
        position: absolute;
        top: 50%;
        left: 50%;
        transform: translate(-50%, -50%);
      }
    }
  }
}
