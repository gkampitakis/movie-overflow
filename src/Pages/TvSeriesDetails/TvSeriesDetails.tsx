import React, { useState, useEffect } from 'react';
import { RouteComponentProps, useHistory } from 'react-router-dom';
import { TvSeriesDetails } from '../../types';
import Placeholder from '../../assets/placeholder.jpeg';
import { Loader } from '../../Components';
import { getTvSeries } from '../../Api';
import './tvSeriesDetails.scss';

const imgSrc = (path: string, width: '185' | '92') =>
  path ? `https://image.tmdb.org/t/p/w${width}${path}` : Placeholder;

export default function _TvSeriesDetails(
  props: RouteComponentProps<{ id: string }>
) {
  const history = useHistory();
  const [loading, setLoading] = useState(false);
  const { id } = props.match.params;
  const [tvSeriesDetails, setTvSeriesDetails] = useState<TvSeriesDetails>();
  const [selectedSeason, setSelectedSeason] = useState(0);

  const handleRequest = () => {
    setLoading(true);

    getTvSeries(id)
      .then((data) => {
        setTvSeriesDetails(data);
      })
      .catch((error) => {
        if (error.status === 404) {
          return history.push({
            pathname: '/notFound'
          });
        } else {
          console.error(error);
        }
      })
      .finally(() => setLoading(false));
  };

  const goTo = (id: string) => {
    history.push({
      pathname: `/person/${id}`
    });
  };

  useEffect(() => {
    handleRequest();
  }, [id]);

  return (
    <section className="tv_series_details">
      {tvSeriesDetails && (
        <>
          <div className="details">
            <header className="tv_header">
              <h1>{tvSeriesDetails.original_name}</h1>
              <p>
                {tvSeriesDetails.genres.map((genre) => genre.name).join(', ') ||
                  'Unknown'}
              </p>
            </header>
            <article className="overview">
              <img
                src={imgSrc(tvSeriesDetails.poster_path, '185')}
                alt={tvSeriesDetails.original_name}
              />
              <div>
                {tvSeriesDetails.tagline && (
                  <p className="tagline">{tvSeriesDetails.tagline}</p>
                )}
                <p>{tvSeriesDetails.overview}</p>
              </div>
            </article>
            <article className="general">
              <div className="released">
                <div>
                  <h3>Status:</h3>
                  <p>{tvSeriesDetails.status}</p>
                </div>
                <div>
                  <h3>Started:</h3>
                  <p>{tvSeriesDetails.first_air_date}</p>
                </div>
                <div>
                  {tvSeriesDetails.last_air_date && (
                    <>
                      <h3>Finished:</h3> <p>{tvSeriesDetails.last_air_date}</p>
                    </>
                  )}
                </div>
                <div>
                  <h3>Seasons: </h3>
                  <p>{tvSeriesDetails.number_of_seasons}</p>
                </div>
                <div>
                  <h3>Vote: </h3>
                  <p>{tvSeriesDetails.vote_average} âœ¨</p>
                </div>
              </div>
            </article>
            {!!tvSeriesDetails.seasons.length && (
              <>
                <hr />
                <h2 className="seasons_title">Seasons</h2>
                <article className="seasons">
                  {tvSeriesDetails.seasons.map((season, idx) => (
                    <div
                      onClick={() => setSelectedSeason(idx)}
                      className={`season ${
                        idx === selectedSeason && 'selected'
                      }`}
                      key={season.id}
                    >
                      <img
                        src={imgSrc(season.poster_path, '92')}
                        alt={season.name}
                      />
                    </div>
                  ))}
                </article>
                <article className="season_details">
                  <h2>
                    {tvSeriesDetails.seasons[selectedSeason].name} | Episodes:{' '}
                    {tvSeriesDetails.seasons[selectedSeason].episode_count}
                  </h2>
                  <p>{tvSeriesDetails.seasons[selectedSeason].overview}</p>
                </article>
              </>
            )}
          </div>
          {(!!tvSeriesDetails.credits.cast.length ||
            !!tvSeriesDetails.credits.crew.length) && (
            <div className="credits">
              {!!tvSeriesDetails.credits.cast.length && <h2>Cast</h2>}
              {tvSeriesDetails.credits.cast.map((person) => (
                <div onClick={() => goTo(person.id)} key={person.id}>
                  <div className="avatar">
                    <img
                      loading="lazy"
                      src={
                        person.profile_path
                          ? imgSrc(person.profile_path, '185')
                          : `https://eu.ui-avatars.com/api/?size=185&name=${person.original_name}`
                      }
                      alt={person.original_name}
                    />
                  </div>
                  <h3>{person.original_name}</h3>
                  <p>as {person.character}</p>
                </div>
              ))}
              {!!tvSeriesDetails.credits.crew.length && <h2>Crew</h2>}
              {tvSeriesDetails.credits.crew.map((person) => (
                <div onClick={() => goTo(person.id)} key={person.id}>
                  <div className="avatar">
                    <img
                      loading="lazy"
                      src={
                        person.profile_path
                          ? imgSrc(person.profile_path, '185')
                          : `https://eu.ui-avatars.com/api/?size=185&name=${person.original_name}`
                      }
                      alt={person.original_name}
                    />
                  </div>
                  <h3>{person.original_name}</h3>
                  <p>{person.job}</p>
                </div>
              ))}
            </div>
          )}
        </>
      )}
      <Loader loading={loading} />
    </section>
  );
}
