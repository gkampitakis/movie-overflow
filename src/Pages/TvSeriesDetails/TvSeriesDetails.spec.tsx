import React from 'react';
import TvSeriesDetails from '.';
import { render, act, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { FetchMock } from 'jest-fetch-mock';
import {
  mockResponseOnce,
  mockDelayedResponseOnce,
  mockHistoryPush,
  tvSeriesDetailsResponse
} from '../../setupTests';

const fetchMock = fetch as FetchMock;
const mockProps = {
  match: { params: { id: 'mockId' } }
} as any;

describe('TvSeriesDetails', () => {
  beforeEach(() => {
    fetchMock.resetMocks();
    mockHistoryPush.mockClear();
    mockResponseOnce(tvSeriesDetailsResponse);
  });

  it('Should show loading', async () => {
    fetchMock.resetMocks();
    mockDelayedResponseOnce({});
    await act(async () => {
      const { container } = render(<TvSeriesDetails {...mockProps} />);

      await screen.findByTestId('loader');

      expect(container).toMatchSnapshot();
    });
  });

  it('Should render consistently', async () => {
    await act(async () => {
      const { container } = render(<TvSeriesDetails {...mockProps} />);

      await screen.findByText(/mockTitle/i);

      expect(container).toMatchSnapshot();
    });
  });

  it("Should go to person's page", async () => {
    await act(async () => {
      render(<TvSeriesDetails {...mockProps} />);

      await screen.findByText(/mockTitle/i);
      const cast = screen.getByAltText('mockName');
      const crew = screen.getByAltText('crewMockName2');

      userEvent.click(cast);
      userEvent.click(crew);

      expect(mockHistoryPush).toHaveBeenCalledWith({
        pathname: '/person/id1'
      });
      expect(mockHistoryPush).toHaveBeenCalledWith({
        pathname: '/person/crewId2'
      });
    });
  });

  it('Should not try to render crew if no resources are returned from api', async () => {
    fetchMock.resetMocks();
    const response = { ...tvSeriesDetailsResponse };
    response.credits.crew = [];
    mockResponseOnce(response);
    await act(async () => {
      const { container } = render(<TvSeriesDetails {...mockProps} />);

      await screen.findByText(/mockTitle/i);

      expect(screen.queryByText('Crew')).toBeNull();
      expect(container).toMatchSnapshot();
    });
  });

  it('Should change selected season', async () => {
    await act(async () => {
      const { container } = render(<TvSeriesDetails {...mockProps} />);

      await screen.findByText(/mockTitle/i);

      const seasonBtn = screen.getByAltText('mockSeason2');
      userEvent.click(seasonBtn);

      expect(screen.queryByText('mockOverview2')).toBeDefined();
      expect(container).toMatchSnapshot();
    });
  });

  it('Should redirect to page not found', async () => {
    fetchMock.resetMocks();
    mockResponseOnce('{}', 404);
    await act(async () => {
      render(<TvSeriesDetails {...mockProps} />);

      setImmediate(() => {
        expect(mockHistoryPush).toHaveBeenCalledWith({
          pathname: '/notFound'
        });
      });
    });
  });

  it('Should log in case of api error', async () => {
    fetchMock.resetMocks();
    mockResponseOnce({ message: 'mockError' }, 400);
    const logSpy = jest.spyOn(console, 'error');

    await act(async () => {
      render(<TvSeriesDetails {...mockProps} />);

      setImmediate(() => {
        expect(logSpy).toHaveBeenCalled();
      });
    });
  });
});
