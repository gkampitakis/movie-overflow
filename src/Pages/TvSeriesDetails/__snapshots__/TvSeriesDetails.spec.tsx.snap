// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TvSeriesDetails Should change selected season 1`] = `
<div>
  <section
    class="tv_series_details"
  >
    <div
      class="details"
    >
      <header
        class="tv_header"
      >
        <h1>
          mockTitle
        </h1>
        <p>
          test
        </p>
      </header>
      <article
        class="overview"
      >
        <img
          alt="mockTitle"
          src="https://image.tmdb.org/t/p/w185/mock_path"
        />
        <div>
          <p
            class="tagline"
          >
            mockTagLine
          </p>
          <p>
            mockOverview
          </p>
        </div>
      </article>
      <article
        class="general"
      >
        <div
          class="released"
        >
          <div>
            <h3>
              Status:
            </h3>
            <p>
              mockStatus
            </p>
          </div>
          <div>
            <h3>
              Started:
            </h3>
            <p />
          </div>
          <div>
            <h3>
              Finished:
            </h3>
             
            <p>
              mockDate
            </p>
          </div>
          <div>
            <h3>
              Seasons: 
            </h3>
            <p>
              10
            </p>
          </div>
          <div>
            <h3>
              Vote: 
            </h3>
            <p>
              100
               ✨
            </p>
          </div>
        </div>
      </article>
      <hr />
      <h2
        class="seasons_title"
      >
        Seasons
      </h2>
      <article
        class="seasons"
      >
        <div
          class="season false"
        >
          <img
            alt="mockSeason1"
            src="https://image.tmdb.org/t/p/w92mockPosterPath1"
          />
        </div>
        <div
          class="season selected"
        >
          <img
            alt="mockSeason2"
            src="https://image.tmdb.org/t/p/w92mockPosterPath2"
          />
        </div>
      </article>
      <article
        class="season_details"
      >
        <h2>
          mockSeason2
           | Episodes:
           
          15
        </h2>
        <p>
          mockOverview2
        </p>
      </article>
    </div>
    <div
      class="credits"
    >
      <h2>
        Cast
      </h2>
      <div>
        <div
          class="avatar"
        >
          <img
            alt="mockName"
            loading="lazy"
            src="https://image.tmdb.org/t/p/w185cast_path"
          />
        </div>
        <h3>
          mockName
        </h3>
        <p>
          as 
          character1
        </p>
      </div>
      <div>
        <div
          class="avatar"
        >
          <img
            alt="mockName2"
            loading="lazy"
            src="https://eu.ui-avatars.com/api/?size=185&name=mockName2"
          />
        </div>
        <h3>
          mockName2
        </h3>
        <p>
          as 
          character2
        </p>
      </div>
    </div>
  </section>
</div>
`;

exports[`TvSeriesDetails Should load default values 1`] = `
<div>
  <section
    class="tv_series_details"
  >
    <div
      class="details"
    >
      <header
        class="tv_header"
      >
        <h1>
          mockTitle
        </h1>
        <p>
          Unknown
        </p>
      </header>
      <article
        class="overview"
      >
        <img
          alt="mockTitle"
          src="placeholder.jpeg"
        />
        <div>
          <p
            class="tagline"
          >
            mockTagLine
          </p>
          <p>
            mockOverview
          </p>
        </div>
      </article>
      <article
        class="general"
      >
        <div
          class="released"
        >
          <div>
            <h3>
              Status:
            </h3>
            <p>
              mockStatus
            </p>
          </div>
          <div>
            <h3>
              Started:
            </h3>
            <p />
          </div>
          <div>
            <h3>
              Finished:
            </h3>
             
            <p>
              mockDate
            </p>
          </div>
          <div>
            <h3>
              Seasons: 
            </h3>
            <p>
              10
            </p>
          </div>
          <div>
            <h3>
              Vote: 
            </h3>
            <p>
              100
               ✨
            </p>
          </div>
        </div>
      </article>
      <hr />
      <h2
        class="seasons_title"
      >
        Seasons
      </h2>
      <article
        class="seasons"
      >
        <div
          class="season selected"
        >
          <img
            alt="mockSeason1"
            src="https://image.tmdb.org/t/p/w92mockPosterPath1"
          />
        </div>
        <div
          class="season false"
        >
          <img
            alt="mockSeason2"
            src="https://image.tmdb.org/t/p/w92mockPosterPath2"
          />
        </div>
      </article>
      <article
        class="season_details"
      >
        <h2>
          mockSeason1
           | Episodes:
           
          10
        </h2>
        <p>
          mockOverview1
        </p>
      </article>
    </div>
    <div
      class="credits"
    >
      <h2>
        Cast
      </h2>
      <div>
        <div
          class="avatar"
        >
          <img
            alt="mockName"
            loading="lazy"
            src="https://image.tmdb.org/t/p/w185cast_path"
          />
        </div>
        <h3>
          mockName
        </h3>
        <p>
          as 
          character1
        </p>
      </div>
      <div>
        <div
          class="avatar"
        >
          <img
            alt="mockName2"
            loading="lazy"
            src="https://eu.ui-avatars.com/api/?size=185&name=mockName2"
          />
        </div>
        <h3>
          mockName2
        </h3>
        <p>
          as 
          character2
        </p>
      </div>
    </div>
  </section>
</div>
`;

exports[`TvSeriesDetails Should not render credits column if no data 1`] = `
<div>
  <section
    class="tv_series_details"
  >
    <div
      class="details"
    >
      <header
        class="tv_header"
      >
        <h1>
          mockTitle
        </h1>
        <p>
          test
        </p>
      </header>
      <article
        class="overview"
      >
        <img
          alt="mockTitle"
          src="https://image.tmdb.org/t/p/w185/mock_path"
        />
        <div>
          <p
            class="tagline"
          >
            mockTagLine
          </p>
          <p>
            mockOverview
          </p>
        </div>
      </article>
      <article
        class="general"
      >
        <div
          class="released"
        >
          <div>
            <h3>
              Status:
            </h3>
            <p>
              mockStatus
            </p>
          </div>
          <div>
            <h3>
              Started:
            </h3>
            <p />
          </div>
          <div>
            <h3>
              Finished:
            </h3>
             
            <p>
              mockDate
            </p>
          </div>
          <div>
            <h3>
              Seasons: 
            </h3>
            <p>
              10
            </p>
          </div>
          <div>
            <h3>
              Vote: 
            </h3>
            <p>
              100
               ✨
            </p>
          </div>
        </div>
      </article>
      <hr />
      <h2
        class="seasons_title"
      >
        Seasons
      </h2>
      <article
        class="seasons"
      >
        <div
          class="season selected"
        >
          <img
            alt="mockSeason1"
            src="https://image.tmdb.org/t/p/w92mockPosterPath1"
          />
        </div>
        <div
          class="season false"
        >
          <img
            alt="mockSeason2"
            src="https://image.tmdb.org/t/p/w92mockPosterPath2"
          />
        </div>
      </article>
      <article
        class="season_details"
      >
        <h2>
          mockSeason1
           | Episodes:
           
          10
        </h2>
        <p>
          mockOverview1
        </p>
      </article>
    </div>
  </section>
</div>
`;

exports[`TvSeriesDetails Should not try to render crew if no resources are returned from api 1`] = `
<div>
  <section
    class="tv_series_details"
  >
    <div
      class="details"
    >
      <header
        class="tv_header"
      >
        <h1>
          mockTitle
        </h1>
        <p>
          test
        </p>
      </header>
      <article
        class="overview"
      >
        <img
          alt="mockTitle"
          src="https://image.tmdb.org/t/p/w185/mock_path"
        />
        <div>
          <p
            class="tagline"
          >
            mockTagLine
          </p>
          <p>
            mockOverview
          </p>
        </div>
      </article>
      <article
        class="general"
      >
        <div
          class="released"
        >
          <div>
            <h3>
              Status:
            </h3>
            <p>
              mockStatus
            </p>
          </div>
          <div>
            <h3>
              Started:
            </h3>
            <p />
          </div>
          <div>
            <h3>
              Finished:
            </h3>
             
            <p>
              mockDate
            </p>
          </div>
          <div>
            <h3>
              Seasons: 
            </h3>
            <p>
              10
            </p>
          </div>
          <div>
            <h3>
              Vote: 
            </h3>
            <p>
              100
               ✨
            </p>
          </div>
        </div>
      </article>
      <hr />
      <h2
        class="seasons_title"
      >
        Seasons
      </h2>
      <article
        class="seasons"
      >
        <div
          class="season selected"
        >
          <img
            alt="mockSeason1"
            src="https://image.tmdb.org/t/p/w92mockPosterPath1"
          />
        </div>
        <div
          class="season false"
        >
          <img
            alt="mockSeason2"
            src="https://image.tmdb.org/t/p/w92mockPosterPath2"
          />
        </div>
      </article>
      <article
        class="season_details"
      >
        <h2>
          mockSeason1
           | Episodes:
           
          10
        </h2>
        <p>
          mockOverview1
        </p>
      </article>
    </div>
    <div
      class="credits"
    >
      <h2>
        Cast
      </h2>
      <div>
        <div
          class="avatar"
        >
          <img
            alt="mockName"
            loading="lazy"
            src="https://image.tmdb.org/t/p/w185cast_path"
          />
        </div>
        <h3>
          mockName
        </h3>
        <p>
          as 
          character1
        </p>
      </div>
      <div>
        <div
          class="avatar"
        >
          <img
            alt="mockName2"
            loading="lazy"
            src="https://eu.ui-avatars.com/api/?size=185&name=mockName2"
          />
        </div>
        <h3>
          mockName2
        </h3>
        <p>
          as 
          character2
        </p>
      </div>
    </div>
  </section>
</div>
`;

exports[`TvSeriesDetails Should render consistently 1`] = `
<div>
  <section
    class="tv_series_details"
  >
    <div
      class="details"
    >
      <header
        class="tv_header"
      >
        <h1>
          mockTitle
        </h1>
        <p>
          test
        </p>
      </header>
      <article
        class="overview"
      >
        <img
          alt="mockTitle"
          src="https://image.tmdb.org/t/p/w185/mock_path"
        />
        <div>
          <p
            class="tagline"
          >
            mockTagLine
          </p>
          <p>
            mockOverview
          </p>
        </div>
      </article>
      <article
        class="general"
      >
        <div
          class="released"
        >
          <div>
            <h3>
              Status:
            </h3>
            <p>
              mockStatus
            </p>
          </div>
          <div>
            <h3>
              Started:
            </h3>
            <p />
          </div>
          <div>
            <h3>
              Finished:
            </h3>
             
            <p>
              mockDate
            </p>
          </div>
          <div>
            <h3>
              Seasons: 
            </h3>
            <p>
              10
            </p>
          </div>
          <div>
            <h3>
              Vote: 
            </h3>
            <p>
              100
               ✨
            </p>
          </div>
        </div>
      </article>
      <hr />
      <h2
        class="seasons_title"
      >
        Seasons
      </h2>
      <article
        class="seasons"
      >
        <div
          class="season selected"
        >
          <img
            alt="mockSeason1"
            src="https://image.tmdb.org/t/p/w92mockPosterPath1"
          />
        </div>
        <div
          class="season false"
        >
          <img
            alt="mockSeason2"
            src="https://image.tmdb.org/t/p/w92mockPosterPath2"
          />
        </div>
      </article>
      <article
        class="season_details"
      >
        <h2>
          mockSeason1
           | Episodes:
           
          10
        </h2>
        <p>
          mockOverview1
        </p>
      </article>
    </div>
    <div
      class="credits"
    >
      <h2>
        Cast
      </h2>
      <div>
        <div
          class="avatar"
        >
          <img
            alt="mockName"
            loading="lazy"
            src="https://image.tmdb.org/t/p/w185cast_path"
          />
        </div>
        <h3>
          mockName
        </h3>
        <p>
          as 
          character1
        </p>
      </div>
      <div>
        <div
          class="avatar"
        >
          <img
            alt="mockName2"
            loading="lazy"
            src="https://eu.ui-avatars.com/api/?size=185&name=mockName2"
          />
        </div>
        <h3>
          mockName2
        </h3>
        <p>
          as 
          character2
        </p>
      </div>
      <h2>
        Crew
      </h2>
      <div>
        <div
          class="avatar"
        >
          <img
            alt="crewMockName"
            loading="lazy"
            src="https://image.tmdb.org/t/p/w185crew_path"
          />
        </div>
        <h3>
          crewMockName
        </h3>
        <p>
          job1
        </p>
      </div>
      <div>
        <div
          class="avatar"
        >
          <img
            alt="crewMockName2"
            loading="lazy"
            src="https://eu.ui-avatars.com/api/?size=185&name=crewMockName2"
          />
        </div>
        <h3>
          crewMockName2
        </h3>
        <p>
          job2
        </p>
      </div>
    </div>
  </section>
</div>
`;

exports[`TvSeriesDetails Should show loading 1`] = `
<div>
  <section
    class="tv_series_details"
  >
    <div
      class="loader"
      data-testid="loader"
    />
  </section>
</div>
`;
